worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/DeviceMonitor.scala

```diff
@@ -225,26 +235,43 @@ class LocalDeviceMonitor(
           try {
             observedDevices.values().asScala.foreach(device => {
               val mountPoints = device.diskInfos.keySet.asScala.toList
-
-              if (checkIoHang && device.ioHang()) {
-                logger.error(s"Encounter device io hang error!" +
-                  s"${device.deviceInfo.name}, notify observers")
-                device.notifyObserversOnNonCriticalError(mountPoints, DiskStatus.IO_HANG)
+              for (concurrentSet <- device.nonCriticalErrors.values().asScala) {
+                for (time <- concurrentSet.asScala) {
+                  if (System.currentTimeMillis() - time > device.notifyErrorExpireTimeout) {
```

Comment:
We'd better extract System.currentTimeMillis() as a variable to avoid the system call each iterator

Reply:
Thanks for pointing out, learned.